lib/AST/DocComment.cpp

```diff
@@ -426,24 +426,44 @@ class CommentProviderFinder final {
     return std::nullopt;
   }
 
+  /// Check if there is an inherited protocol that has a default implementation
+  /// of `VD` with a doc comment.
   std::optional<ResultWithDecl> findDefaultProvidedDecl(const ValueDecl *VD) {
-    // Only applies to protocol extension member.
-    auto *protocol = VD->getDeclContext()->getExtendedProtocolDecl();
-    if (!protocol)
+    NominalTypeDecl *nominalType =
+        dyn_cast_or_null<NominalTypeDecl>(VD->getDeclContext()->getAsDecl());
+    if (!nominalType) {
+      nominalType = VD->getDeclContext()->getExtendedProtocolDecl();
+    }
+    if (!nominalType)
       return std::nullopt;
 
     SmallVector<ValueDecl *, 2> members;
-    protocol->lookupQualified(const_cast<ProtocolDecl *>(protocol),
-                              DeclNameRef(VD->getName()),
-                              VD->getLoc(), NLOptions::NL_ProtocolMembers,
-                              members);
+    nominalType->lookupQualified(nominalType, DeclNameRef(VD->getName()),
+                                 VD->getLoc(), NLOptions::NL_ProtocolMembers,
+                                 members);
 
     std::optional<ResultWithDecl> result;
+    Type vdComparisonTy = VD->getInterfaceType();
+    if (!vdComparisonTy) {
+      return std::nullopt;
+    }
+    if (auto fnTy = vdComparisonTy->getAs<AnyFunctionType>()) {
+      // Strip off the 'Self' parameter.
+      vdComparisonTy = fnTy->getResult();
+    }
```

Comment:
Wonder if we ought to have a utility for removing the curried self, could be used in a couple of other places:

https://github.com/apple/swift/blob/15fd443be60cfb52852c853c4fa0d6a1622ada10/lib/Sema/TypeCheckProtocol.cpp#L4894-L4896

https://github.com/apple/swift/blob/15fd443be60cfb52852c853c4fa0d6a1622ada10/lib/Sema/AssociatedTypeInference.cpp#L1990-L1993

https://github.com/apple/swift/blob/15fd443be60cfb52852c853c4fa0d6a1622ada10/lib/AST/Decl.cpp#L4937-L4939